src/IssueViz/Models/QuickFixVisualization.cs

```diff
@@ -29,38 +32,69 @@ public interface IQuickFixVisualization
         IQuickFix Fix { get; }
 
         IReadOnlyList<IQuickFixEditVisualization> EditVisualizations { get; }
+
+        bool IsApplicable(ITextSnapshot currentSnapshot);
     }
 
     internal class QuickFixVisualization : IQuickFixVisualization
     {
+        private readonly ISpanTranslator spanTranslator;
+
         public QuickFixVisualization(IQuickFix fix, IReadOnlyList<IQuickFixEditVisualization> editVisualizations)
+            : this(fix, editVisualizations, new SpanTranslator())
+        {
+            Fix = fix;
+            EditVisualizations = editVisualizations;
+        }
+
+        internal QuickFixVisualization(IQuickFix fix,
+            IReadOnlyList<IQuickFixEditVisualization> editVisualizations,
+            ISpanTranslator spanTranslator)
         {
+            this.spanTranslator = spanTranslator;
             Fix = fix;
             EditVisualizations = editVisualizations;
         }
 
         public IQuickFix Fix { get; }
 
         public IReadOnlyList<IQuickFixEditVisualization> EditVisualizations { get; }
+
+        public bool IsApplicable(ITextSnapshot currentSnapshot) =>
```

Comment:
I decided to put it on the Visualization class itself since it made the mocking very easy in the rest of the code.
A quick fix now exposes IsApplicable method and the underlying implementation checks if all the edits still have the same text.